db/migrate/20230719080900_fix_ubuntu_versions.rb

```diff
@@ -0,0 +1,13 @@
+class FixUbuntuVersions < ActiveRecord::Migration[6.0]
+  def up
+    # For every Ubuntu operating system with a set minor version we
+    #   * Check if there is already an operating system present with the same major and minor version combined in the major field
+    #   * If not, we update the version
+    Operatingsystem.where('name ~* :name', name: 'ubuntu')
+                    .merge(Operatingsystem.where.not(minor: [nil, ''])).each do |os|
+      if !Operatingsystem.where('name ~* :name', name: 'ubuntu').merge(Operatingsystem.where(major: "#{os.major}.#{os.minor}")).merge(Operatingsystem.where(minor: [nil, ''])).first
```

Comment:
I'm not familiar with `.merge` and the ORM. Does that translate to something in SQL or is it doing 3 queries for every OS?

Reply:
The merge method can be used to combine two Active Record relations into one. I used it to combine the case insensitive regex search with the other parameters. But I just noticed I could have combined the two calls to `merge` into one.

But if I can rely on the normalized OS name "Ubuntu" both queries in line 6/7 and line 8 get really simple. I adjusted them accordingly.